lib/babe/parachain_inherents_test.go

```diff
@@ -0,0 +1,295 @@
+		t.Run(c.name, func(t *testing.T) {
+			validityAttestation := NewValidityAttestation()
+			err := validityAttestation.Set(c.enumValue)
+			require.NoError(t, err)
+
+			bytes, err := scale.Marshal(validityAttestation)
+			require.NoError(t, err)
+
+			fmt.Printf("bytes 0x%x\n", bytes)
+			require.Equal(t, c.encodingValue, bytes)
+
+			newDst := NewValidityAttestation()
+			err = scale.Unmarshal(bytes, &newDst)
+			require.NoError(t, err)
+
+			if !reflect.DeepEqual(validityAttestation, newDst) {
+				panic(fmt.Errorf("uh oh: \n%+v \n\n%+v", validityAttestation, newDst))
+			}
+		})
+	}
+
+}
+
+func TestParachainInherents(t *testing.T) {
+	expectedParaInherentsbytes := []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}                                            //nolint:lll
+	expectedInherentsBytes := []byte{4, 112, 97, 114, 97, 99, 104, 110, 48, 149, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0} //nolint:lll
+	// corresponding rust struct
+	// ----------------------------------------
+	// let para_int: polkadot_primitives::v2::InherentData = polkadot_primitives::v2::InherentData {
+	// 	bitfields: Vec::new(),
+	// 	backed_candidates: Vec::new(),
+	// 	disputes: Vec::new(),
+	// 	parent_header: polkadot_core_primitives::Header {
+	// 	   number: 0,
+	// 	   digest: Default::default(),
+	// 	   extrinsics_root: Default::default(),
+	// 	   parent_hash: Default::default(),
+	// 	   state_root : Default::default(),
+	//    },
+	// };
+	// ----------------------------------------
+	// way to get inherents encoding from rust
+	// ----------------------------------------
+	// let mut inherents: sp_inherents::InherentData = sp_inherents::InherentData::new();
+	// inherents.put_data(*b"parachn0", &para_int).unwrap();
+	// println!("{:?}", inherents.encode());
+
+	parachainInherent := ParachainInherentData{
+		Bitfields:        []UncheckedSignedAvailabilityBitfield{},
+		BackedCandidates: []BackedCandidate{},
+		Disputes:         MultiDisputeStatementSet{},
+		ParentHeader: types.Header{
+			ParentHash:     common.Hash{},
+			Number:         0,
+			StateRoot:      common.Hash{},
+			ExtrinsicsRoot: common.Hash{},
+			Digest:         scale.VaryingDataTypeSlice{},
+		},
+	}
+
+	actualParaInherentBytes, err := scale.Marshal(parachainInherent)
+	require.NoError(t, err)
+
+	require.Equal(t, len(expectedParaInherentsbytes), len(actualParaInherentBytes))
+	require.True(t, bytes.Equal(actualParaInherentBytes, expectedParaInherentsbytes))
```

Comment:
nit same thing simpler

```suggestion
	assert.Equal(t, expectedParaInherentsbytes, actualParaInherentBytes)
```